kroxylicious-kms/src/main/java/io/kroxylicious/kms/service/UnsupportedRequestException.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+/**
+ * Thrown when a http request fails.
+ */
+public class UnsupportedRequestException extends KmsException {
```

Comment:
Introducing a specialization of KmsException is a good idea.  However, the name is not good.   Your use-case for is when the request fails for an unclassified reason.  So, the name should reflect that.  UnsupportedRequestException is misleading because it makes it sounds like the server is rejecting an unsupported request.

Reply:
Removed as is causing more troubles than leaving the `IllegalStateException` as it was. ;)